File: 90-subsets-ii/subsets-ii.cpp
class Solution {
  public:
  
    void loki(vector<int> &arr, int idx, vector<int> &curr, set<vector<int>> &ans ){
        if(idx == arr.size()) {
            ans.insert(curr);
            return;
        }
        loki(arr, idx+1, curr, ans);
        curr.push_back(arr[idx]);
        loki(arr, idx+1, curr,ans);
        curr.pop_back();
        
    }
    vector<vector<int>> subsetsWithDup(vector<int>& arr) {
        vector<int>curr;
        sort(arr.begin(), arr.end());
        set<vector<int>> ans;
        loki(arr,0, curr, ans);
        vector<vector<int>> ans1;
        for(auto i : ans) ans1.push_back(i);
        return ans1;
    }
};

File: 90-subsets-ii/subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        ans = []
        subset = []
        nums.sort()
        def back(idx):
            ans.append(subset.copy())
            for i in range(idx, len(nums)):
                if i!=idx and nums[i]==nums[i-1]: continue
                subset.append(nums[i])
                back(i+1)
                subset.pop()
        back(0)
        return ans